### PR TITLE
Kentucky adjusted gross income

### LINKED ISSUE #3081: Kentucky adjusted gross income
Reference: https://revenue.ky.gov/Forms/Schedule%20M%202022.pdf

### PR DESCRIPTION
**Pull request by WeihaoZ724**
Fixes #3081

### Summary
:memo::heavy_plus_sign::link:

This file adds the `subtractions` parameter for Kentucky income tax, which defines the income types that can be deducted from the federal AGI. This is part of a pull request to implement the AGI calculation for Kentucky.

> _`subtractions` list_
> _deducts income for state tax_
> _autumn leaves fall fast_

### Walkthrough
* Add a parameter for subtractions from federal AGI for Kentucky income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/3082/files?diff=unified&w=0#diff-ea532d299f6b36f3fffc725241761be7cb05f236973650339a2cad3c12b7e261R1-R17))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+description: Kentucky counts these sources as subtractions.
+values:
+  2021-01-01:
+  # - Kentucky income tax refund
+    - us_govt_interest
+    - ky_pension_income_exclusion  #TODO: merge pension exlsuion PR before merging AGI
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml:6]: it was merged, please add the variable to this list and tix tests errors

File: policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  # - Kentucky income tax refund
+    - us_govt_interest
+    - ky_pension_income_exclusion  
+    - taxable_social_security
+  # - Resident adjustment for partnerships
+  # - Kentucky depreciation 
+metadata:
+  unit: list
+  reference:
+    - title: Kentucky Statutes 141.019 (1)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml:13]:
```suggestion
    - title: Kentucky Statutes 141.019 
```

File: policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+description: Kentucky counts these sources as subtractions.
+values:
+  2021-01-01:
+  # - Kentucky income tax refund
+    - us_govt_interest
+    - ky_pension_income_exclusion  
+    - taxable_social_security
+  # - Resident adjustment for partnerships
+  # - Kentucky depreciation
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml:9]:
```suggestion
  # - Kentucky depreciation 
     - military_service_income
```

File: policyengine_us/variables/gov/states/ky/tax/income/ky_agi.py [hunk 4]
```diff
@@ -9,3 +9,5 @@ class ky_agi(Variable):
     definition_period = YEAR
     reference = "https://revenue.ky.gov/Forms/740%20Packet%20Instructions%205-9-23.pdf#page=11"
     defined_for = StateCode.KY
+    adds = ["ky_additions"]
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ky/tax/income/ky_agi.py:12]:
```suggestion
    adds = ["ky_additions", "adjusted_gross_income"]
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+description: Kentucky counts these sources as subtractions.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ky/tax/income/subtractions.yaml:1]:
```suggestion
description: Kentucky subtracts these sources from adjusted gross income.
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**